layers/core_validation.cpp

```diff
@@ -1622,7 +1630,7 @@ bool CoreChecks::ValidatePipelineLocked(std::vector<std::shared_ptr<PIPELINE_STA
         }
 
         // Validate polygon mode
-        auto raster_state_ci = create_info.pRasterizationState;
+        auto raster_state_ci = pipeline->RasterizationState();
```

Comment:
auto *

Reply:
I don't think this is necessary for pointers. If a pointer type is returned, `auto` is deduced as a pointer. I think the issue is references, which `auto` will never do implicitly. e.g., https://godbolt.org/z/7TWT9Tzah

Notice the types for a1 and a2 are equivalent (namely, `Pi` meaning "pointer to int").